routing/rt/rt.go

```diff
@@ -0,0 +1,324 @@
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	newChannels := rt2.AllChannels()
+	for _, channel := range newChannels {
+		if !rt.HasChannel(channel.Id1, channel.Id2, channel.EdgeID) && IsClose(channel.Id1, channel.Id2) {
+			rt.AddChannel(channel.Id1, channel.Id2, channel.EdgeID, channel.Info)
+		}
+	}
+}
+
+func (rt *RoutingTable) ApplyDiffBuffFiltered(diffBuff *DifferenceBuffer, Node graph.ID, MaxDist int) {
+	rtCopy := rt.Copy()
+	rtCopy.ApplyDiffBuff(diffBuff)
+	distances, _, err := rtCopy.G.Bfs(Node)
+	if err != nil {
+		panic(err)
+	}
+	IsClose := func(u, v graph.ID) bool {
+		distu, ok1 := distances[u]
+		distv, ok2 := distances[v]
+		if !ok1 || !ok2 {
+			return false
+		}
+		return (distu + distv) <= 2*MaxDist
+	}
+	filteredDiffBuff := NewDifferenceBuffer()
+	for _, chOp := range *diffBuff {
+		if chOp.Operation == REMOVE_CHANNEL || (chOp.Operation == ADD_CHANNEL && IsClose(chOp.Id1, chOp.Id2)) {
+			*filteredDiffBuff = append(*filteredDiffBuff, chOp)
+		}
+	}
+	rt.ApplyDiffBuff(filteredDiffBuff)
+}
+
+func AddressDistance(x1, x2 graph.ID) int {
+	// TODO: INSERT CORRECT FUNCTION. It is important only for beacon-related functionality.
+	return 1
+}
+
+func (rt *RoutingTable) ClosestAddressNode(dest graph.ID, exclude map[graph.ID]struct{}) *graph.ID {
+	found := false
+	var x graph.ID
+	var nodeId graph.ID
+	g := rt.G
+	for _, nodeId = range g.GetIDs() {
+		if exclude != nil {
+			if _, ok := exclude[nodeId]; ok {
+				continue
+			}
+		}
+		if !found || AddressDistance(dest, x) > AddressDistance(dest, nodeId) {
+			x = nodeId
+			found = true
+		}
+	}
+	if found {
+		return &x
+	} else {
+		return nil
+	}
+}
+
+func (rt *RoutingTable) String() string {
+	rez := ""
+	edges := rt.G.GetUndirectedEdges()
+	for _, edge := range edges {
+		rez += fmt.Sprintf("%v %v %v", edge.Source(), edge.Target(), edge.Info())
+	}
+	return rez
+}
+
+// Find k-shortest path between source and destination
+func (rt *RoutingTable) KSP(src, dst graph.ID, k int) ([][]graph.ID, error) {
+	return graph.KSP(rt.G, src, dst, k)
+}
+
+// Returns copy of channel info for channel in routing table
+func (rt *RoutingTable) GetChannelInfo(id1, id2 graph.ID, edgeId graph.EdgeID) (*graph.ChannelInfo, error) {
+	info, err := rt.G.GetInfo(id1, id2, edgeId)
+	if err != nil {
+		return nil, err
+	}
+	return info.Copy(), nil
+}
+
+// Marshall into object. Difference buffers are NOT marshalled.
+func (rt *RoutingTable) Marshall(w io.Writer) error {
+	encoder := gob.NewEncoder(w)
+	err := encoder.Encode(rt.G)
+	return err
+}
+
+// Unmarshall object. Difference buffers are NOT marshalled.
+func UnmarshallRoutingTable(r io.Reader) (*RoutingTable, error) {
+	decoder := gob.NewDecoder(r)
+	rt1 := NewRoutingTable()
+	err := decoder.Decode(rt1.G)
+	if err != nil {
+		return nil, err
+	} else {
+		return rt1, nil
+	}
+}
+
+func init() {
```

Comment:
This init function is no longer needed as we now have (after my review comments are addressed)  manual serialization of the relevant objects in the `lnwire` package.